src/components/transport_manager/src/usb/usb_device_scanner.cc

```diff
@@ -171,14 +173,14 @@ class AoaInitSequence::AoaTurnIntoAccessoryMode : public UsbControlOutTransfer {
   }
 };
 
-static char manufacturer[] = "SDL";
-static char model_name[] = "Core";
-static char description[] = "SmartDeviceLink Core Component USB";
-static char version[] = "1.0";
-static char uri[] = "http://www.smartdevicelink.org";
-static char serial_num[] = "N000000";
-
-AoaInitSequence::AoaInitSequence() : UsbControlTransferSequence() {
+AoaInitSequence::AoaInitSequence(const TransportManagerSettings& settings)
+    : UsbControlTransferSequence() {
+  auto manufacturer = settings.aoa_filter_manufacturer().c_str();
+  auto model_name = settings.aoa_filter_model_name().c_str();
+  auto description = settings.aoa_filter_description().c_str();
+  auto version = settings.aoa_filter_version().c_str();
+  auto uri = settings.aoa_filter_uri().c_str();
+  auto serial_num = settings.aoa_filter_serial_number().c_str();
   AddTransfer(new AoaGetProtocolRequest);
   AddTransfer(new AoaSendIdString(0, manufacturer, sizeof(manufacturer)));
```

Comment:
instead of recounting the string with sizeof, why not use c_str() and size() in the AddTransfer calls?